tests/test_plot.py

```diff
@@ -623,3 +623,31 @@ def pdf(x, a=1 / np.sqrt(2 * np.pi), x0=0, sigma=1, offset=0):
     assert h.plot_pull(pdf, fit_fmt=r"{name} = {value:.3g} $\pm$ {error:.3g}")
 
     return fig
+
+
+@pytest.mark.mpl_image_compare(baseline_dir="baseline", savefig_kwargs={"dpi": 50})
+def test_plot1d_auto_handling():
+    """
+    Test plot() by comparing against a reference image generated via
+    `pytest --mpl-generate-path=tests/baseline`
+    """
+
+    np.random.seed(42)
+
+    h = Hist(
+        axis.Regular(10, 0, 10, name="variable", label="variable"),
+        axis.StrCategory("", name="dataset", growth=True),
```

Comment:
Does this support being used with unnamed axes?

Reply:
I'll add it, I think before the changes you proposed it did not

Reply:
Okay, after that, I think this is ready.